### PR TITLE
[TASK] Make compatible with 10

### LINKED ISSUE #17: Outdated usage of TsConfig
Since 9 the tsconfig usage is deprecated, see https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/9.3/Deprecation-84993-DeprecateSomeTSconfigRelatedMethods.html

with 10 this doesn't work at all

### PR DESCRIPTION
**Pull request by georgringer**
Fixes #17

### CODE REVIEW

File: Classes/Hooks/DefaultUploadFolder.php [hunk 1]
```diff
@@ -52,54 +44,12 @@ public function getDefaultUploadFolder($params, BackendUserAuthentication $backe
...
-    )
-    {
-        $subFolder = $defaultPageTs['default_upload_folders.'][$table] ?? '';
-        if (empty($subFolder)) {
-            $subFolder = $userTsConfig['default_upload_folders.'][$table] ?? '';
-        }
-        return $subFolder;
-    }
-
-    protected function getDefaultUploadFolderForAllTables(
```
Comment (frans-beech-it) [inline Classes/Hooks/DefaultUploadFolder.php:95]: @georgringer you also removed the `defaultForAllTables` fallback

**Comment by frans-beech-it** [comment]
Tnx @georgringer

**Comment by tbsschmdt** [comment]
I believe this change was unnecessary because `BackendUserAuthentication->getTSConfig()` is not deprecated or removed in TYPO3 10. See: https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/9.3/Deprecation-84993-DeprecateSomeTSconfigRelatedMethods.html

Unfortunately this change introduced new problems. First there must be no return type declaration for method `getDefaultUploadFolder` as `$params['uploadFolder']` is empty if the backend user has no file mounts at all. Second there is a check if `$subFolder` is an instance of `Folder` which is never the case. `$subFolder` is always a string.

I would appreciate if this change could be reverted.

**Comment by tbsschmdt** [comment]
I created a pull request to revert these changes: #24
(In my previous comment I mentioned that there is a check if `$subfolder` is an instance of `Folder`. This is not true. I somehow misread the code.)

**Comment by fsaris** [comment]
Tnx for the PR. Will try to check it soon.

@georgringer do you remember why you wanted to remove it completely?

**Comment by georgringer** [comment]
Not Really

**Comment by tbsschmdt** [comment]
Looking at #17 I guess it was a misunderstanding of [Deprecation: #84993 - Deprecate some TSconfig related methods](https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/9.3/Deprecation-84993-DeprecateSomeTSconfigRelatedMethods.html)
Some TSconfig methods have been deprecated but not `TYPO3\CMS\core\Authentication\BackendUserAuthentication->getTSConfig()` which is used in _beechit/default-upload-folder_.

**Pull request merged**